File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {

    public int checkPalindrome(String s, int l, int r, int res){
        res = 0;
        while (l>=0 && r<s.length() && s.charAt(l)==s.charAt(r)){
                res++;
                l--;
                r++;
            }
        return res;
    }

    public int countSubstrings(String s) {
        int res = 0;
        for(int i=0; i<s.length(); i++){
            res += checkPalindrome(s, i, i, res);
            res += checkPalindrome(s, i, i + 1, res);
        }
        return res;
    }
}

File: 0647-palindromic-substrings/0647-palindromic-substrings.py
class Solution:
    
    def checkPalindrome(self, s, l, r, res):
        res = 0
        while l>=0 and r<len(s) and s[l] == s[r]:
            l-=1
            r+=1
            res+=1
        return res

    def countSubstrings(self, s: str) -> int:
        res = 0
        for i in range(len(s)):
            res += self.checkPalindrome(s, i, i, res)
            res += self.checkPalindrome(s, i, i+1, res)
        return res
        